lib/valueflow.cpp

```diff
@@ -4214,6 +4243,32 @@ static bool hasContainerSizeGuard(const Token *tok, unsigned int containerId)
     return false;
 }
 
+static bool isContainerSize(const Token* tok)
+{
+    if (!Token::Match(tok, "%var% . %name% ("))
+        return false;
+    if (!astIsContainer(tok))
+        return false;
+    if (tok->valueType()->container && tok->valueType()->container->getYield(tok->strAt(2)) == Library::Container::Yield::SIZE)
+        return true;
+    if (Token::Match(tok->tokAt(2), "size|length ( )"))
```

Comment:
I am not sure I see the point to have this. If you want to have it, I feel a motivation is needed.

If `tok` is a container then there must be library configuration, unless I am mistaken.

I am in general not feeling comfortable about having hardcoded knowledge in the code.

Reply:
> If tok is a container then there must be library configuration, unless I am mistaken.

There isn't a library configuration for `c` even though `astIsContainer` is true:

```cpp
void a() {
  std::string b[];
  for (auto c : b)
    c.data();
}
```

> I am in general not feeling comfortable about having hardcoded knowledge in the code.

It tries to use the library, but if thats not available then it assumes the container is a C++ container(or string) which seems reasonable.